### PR TITLE
feat: add redis and mongo forms

### LINKED ISSUE #880: Form - Health - Mongo

### LINKED ISSUE #881: Form - Health - Redis

### PR DESCRIPTION
**Pull request by yashmehrotra**
Fixes #880 and Fixes #881

### CODE REVIEW

File: src/components/Forms/Health/RedisHealthFormEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+      <FormikTextInput name={`${fieldName}.description`} label="Description" />
+
+      <FormikTextInput
+        name={`${fieldName}.addr`}
+        label="Address"
+        required
+        hint="host:port"
+      />
+      <FormikTextInput
```
Comment (mainawycliffe) [inline src/components/Forms/Health/RedisHealthFormEditor.tsx:56]: Quick question: Is this supposed to be a number field only? 

image

File: src/components/Forms/Health/RedisHealthFormEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          name={`${fieldName}.icon`}
+          label="Icon"
+        />
+      </div>
+
+      <FormikScheduleField name={`${name}.schedule`} />
+
+      <FormikTextInput name={`${fieldName}.description`} label="Description" />
+
+      <FormikTextInput
```
Comment (mainawycliffe) [inline src/components/Forms/Health/RedisHealthFormEditor.tsx:50]: May be make the hint more descriptive or give an example.

image

**Review by mainawycliffe — COMMENTED** [review]
Just a couple of nitpicking comments otherwise looks good.

File: src/components/Forms/Health/RedisHealthFormEditor.tsx (hunk 1 above)
Comment (yashmehrotra) [reply src/components/Forms/Health/RedisHealthFormEditor.tsx:56]: Yeah, redis databases are numbers

File: src/components/Forms/Health/RedisHealthFormEditor.tsx (hunk 2 above)
Comment (yashmehrotra) [reply src/components/Forms/Health/RedisHealthFormEditor.tsx:50]: Can you suggest a better description ?

File: src/components/Forms/Health/MongoHealthFormEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          className="flex flex-col w-1/2"
+          name={`${fieldName}.icon`}
+          label="Icon"
+        />
+      </div>
+
+      <FormikScheduleField name={`${name}.schedule`} />
+
+      <FormikTextInput name={`${fieldName}.username`} label="Username" />
+      <FormikTextInput name={`${fieldName}.password`} label="Password" />
```
Comment (moshloop) [inline src/components/Forms/Health/MongoHealthFormEditor.tsx:49]: These should be env var fields + we need a connection selection field

Comment (moshloop) [reply src/components/Forms/Health/MongoHealthFormEditor.tsx:49]: Also missing the url field

Comment (mainawycliffe) [reply src/components/Forms/Health/MongoHealthFormEditor.tsx]: @moshloop for the connection field, is it a dropdown connection to the connection view?

**Comment by moshloop** [comment]
I think the required field on the db should be removed, if not specified its a 0

File: src/components/Forms/Health/RedisHealthFormEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        required
+        hint="Address to redis server in the format of host:port"
+      />
+
+      <FormikTextInput
+        name={`${fieldName}.db`}
+        label="Database"
+        type="number"
+        hint="Database to be selected after connecting to the server"
+        required
```
Comment (moshloop) [inline src/components/Forms/Health/RedisHealthFormEditor.tsx:62]:
```suggestion
```

**Comment by moshloop** [comment]
@mainawycliffe on the Redis side the database should default to `0` and not be required
On the mongo side we have duplicated user/pass

We need to update the Auth component so that NTLM and NTLM2 optional e.g. only specified on the http check as:

```jsx
<Auth 
   types={{[
  {label: "NTLM",  value: {ntlm: true}},
  {label: "NTLMv2",  value: {ntlm2: true}},
]}}
```

We might then reuse it like:


```jsx
<Auth 
   types={{[
  {label: "Scram",  value: {authType: "scram"}},
  {label: "Digest",  value: {authType: "Digest"}},
]}}
```

For the EnvVarValue selector we also need options to `hideName` and `hideCheckbox` which would then render as:

```

(When Value is selected)
Username:   [Value] [Secret] [Configmap] 
                    <input> 
(When config/secret selected)
Username:   [Value] [Secret] [Configmap] 
                 Name: <input> Key: <input>       
```

**Comment by mainawycliffe** [comment]
> For the EnvVarValue selector we also need options to `hideName` and `hideCheckbox` which would then render as:
> 
> ```
> 
> (When Value is selected)
> Username:   [Value] [Secret] [Configmap] 
>                     <input> 
> (When config/secret selected)
> Username:   [Value] [Secret] [Configmap] 
>                  Name: <input> Key: <input>     
> ```

@moshloop I don't understand what you mean here?

**Comment by mainawycliffe** [comment]
@moshloop is this what you had in mind?

**Comment by moshloop** [comment]
@mainawycliffe - I think lets rather do:
Username:   
 <**Static**>|<Config Map>|<Secret>.  [ _______________ ]
Password:
Name  [ _______________ ] Key  [ _______________ ]  <Static>|<Config Map>|<**Secret**>

- These changes should also apply to all other auth fields 
- NTLM and NTLMv2 are only application to HTTP and should be hidden for others with something like:

```
<Auth 
   types={{[
  {label: "NTLM",  value: {ntlm: true}},
  {label: "NTLMv2",  value: {ntlm2: true}},
]}}
```

**Comment by moshloop** [comment]
@mainawycliffe why do we have checkboxes now? It should be Label, radiobuttongroup, fields ?

**Comment by moshloop** [comment]
@mainawycliffe we still seem to have a scrolling issue on the modal header ?

**Comment by mainawycliffe** [comment]
@moshloop the modal scrolls on the health page should be fixed now

**Pull request merged**